### PR TITLE
Additional printer columns for crd

### PR DESCRIPTION
**Pull request by RafikFarhad**
Closes: #7633

Hello,
This is my first PR. Help me find where I need to write tests for this addition.

------
The CRD objects are shown with `additionalPrinterColumns`.

image

### CODE REVIEW

**Review by maciaszczykm — COMMENTED** [review]
Good idea. Looks good to me. We would need to test it.

@floreks Any thoughts?

**Comment by RafikFarhad** [comment]
@maciaszczykm I have attempted to write a test, but I am blocked. Can you please suggest some guidelines here?

File: modules/api/pkg/resource/customresourcedefinition/v1/detail_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 The Kubernetes Authors.
```
Comment (Jont828) [inline modules/api/pkg/resource/customresourcedefinition/v1/detail_test.go:1]:
```suggestion
// Copyright 2023 The Kubernetes Authors.
```

File: modules/api/pkg/resource/customresourcedefinition/types/types.go [hunk 2]
```diff
@@ -59,31 +59,48 @@ type CustomResourceDefinitionDetail struct {
 	Errors []error `json:"errors"`
 }
 
+type AdditionalPrinterColumn struct {
+	Name     string `json:"name"`
+	Type     string `json:"type,omitempty"`
+	Priority int32  `json:"priority,omitempty"`
+	JSONPath string `json:"jsonPath"`
```
Comment (Jont828) [inline modules/api/pkg/resource/customresourcedefinition/types/types.go:66]: Maybe we can add a description field as shown [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#additional-printer-columns) that can pop up when you hover over the column name?

**Pull request closed without merging**